packages/paste-core/components/button/src/index.tsx

```diff
@@ -30,14 +30,14 @@ import {ResetButton} from './ResetButton';
 const AnimatedBox = animated(Box);
 
 // If size isn't passed, come up with a smart default:
-// - 'reset' for variant 'link'
+// - 'reset' for reset
 // - 'icon' if there's 1 child that's an icon
 // - 'default' otherwise
 const getButtonSize = (variant: ButtonVariants, children: React.ReactNode, size?: ButtonSizes): ButtonSizes => {
   let smartSize: ButtonSizes = 'default';
   if (size != null) {
     smartSize = size;
-  } else if (variant === 'link' || variant === 'destructive_link' || variant === 'reset') {
+  } else if (variant === 'reset') {
```

Comment:
So the reason you can't use the default size for links is that default sizing comes with padding. 

![image](https://user-images.githubusercontent.com/368249/153052107-b8c1aa38-4beb-4af6-8f14-3a7a6a3d27c9.png)

As I mentioned in Slack, the real fix is to make sure reset, at the very least _sets a font size and line height_ by explicitly inheriting them from their surrounding text.

Which I suspect is exactly the behaviour you would get from an Anchor component too. It inherits the parent font and line height sizes

Reply:
I see, I see. I did see an inherit font size property, but that was exclusively on the ResetButton component, and not in the broader base styles for the size prop. Adjusted the code, hopefully this is closer to what you had in mind.